mne/tests/test_epochs.py

```diff
@@ -3357,3 +3359,34 @@ def test_empty_constructor():
     event_id = 1
     tmin, tmax, baseline = -0.2, 0.5, None
     BaseEpochs(info, None, None, event_id, tmin, tmax, baseline)
+
+
+def fun(data_epochs):
+    """Auxiliary function to test_apply function."""
+    matrix = np.random.rand(10, 10)
+    return matrix @ data_epochs
+
+
+def test_apply_function():
+    """Test epochs takes into account repeated events."""
+    data = np.random.rand(2, 10, 1000)
+    events = np.array([[0, 0, 1], [INT32_MAX, 0, 2]])
+    info = mne.create_info(10, 1000., 'eeg')
+    epochs = mne.EpochsArray(data, info, events)
+    data_epochs = epochs.get_data()
+    # check apply_function in all channels at the time
+    out = epochs.apply_function(fun, channel_wise=False)
+    assert np.shape(out) == np.shape(data_epochs)
+    # check apply_function channel-wise
+    # let's filter each channel
+    filter_params = dict(sfreq=info['sfreq'], l_freq=10, h_freq=12,
+                         l_trans_bandwidth=4, h_trans_bandwidth=4)
+    out_ch = epochs.apply_function(filter_data, channel_wise=True,
+                                   **filter_params)
+    out_ch_data = out_ch._data
+    # checking if the filtering was correctly applied
+    psd_out, freqs = psd_array_welch(out_ch_data[0, 0, :], sfreq=info['sfreq'])
+    freqs_up_out = freqs[psd_out > 1e-6]
+    psd_out_raw, freqs = psd_array_welch(data[0, 0, :], sfreq=info['sfreq'])
+    freqs_up_raw = freqs[psd_out_raw > 1e-6]
+    assert len(freqs_up_out) < len(freqs_up_raw)
```

Comment:
this seems unnecessarily complex way to test `apply_function()`.  Why not a super simple function that just flips the sign of every data value?  Easy to understand, easy to test.